lib/orogen/gen/tasks.rb

```diff
@@ -533,6 +533,20 @@ def generate
                     hidden_operation("__orogen_getTID", "    #ifdef HAS_GETTID\nreturn syscall(SYS_gettid);\n#else\nreturn 0;\n#endif").
                         returns("int").
                         doc("returns the PID for this task")
+
+                    if extended_state_support?
```

Comment:
> If using metadata, I would add a convenience function in the cpp_proxies for getting the 
data out. I don't see, that we would win much by it. 

Well... You would put a convenience method in orocos_cpp_base, no ?

> If it is only about compile time, we could move all base operations into a superclass and
derive all orogen generated Tasks from it.

It's also about storing metadata where it belongs. About having a superclass, I start to wonder if it is not the best course of action anyways. However, the problem is that it would break using orogen for non-Rock users. Will have to think about it.